src/view/states/task_input/AddTaskState.h

```diff
@@ -6,19 +6,24 @@
 #define SIMPLETASKMANAGER_ADDTASKSTATE_H
 
 #include "states/State.h"
-#include "states/factory/StatesFactory.h"
+#include "view/commands/factory/CommandFactory.h"
 #include "user_interface/StateContext.h"
+#include "view/user_interface/ConsoleStateMachine.h"
 
 #include <memory>
 #include <optional>
 
 class AddTaskState : public State {
 public:
-    explicit AddTaskState(const std::shared_ptr<StatesFactory>& factory);
+    AddTaskState(StateType next_state,
```

Comment:
AddTaskState, as any, could carry error_state in general - what if the task was not added? (task id collision, network issues, permissions, etc)

Reply:
Each state's responsibility is ask user for information and create a `command`. Processing of errors you've mentioned isn't responsibility of states.